### ISSUE TITLE
How do build

### ISSUE THREAD

**Issue by MetaHubbe** [GabeRundlett/desktop-shadertoy#17]
Build instructions would be nice.

**Comment by Agamenon-sama** [comment]
I agree that this should be added to the `README.md` file but until then, here is how to do it:

## Configure the build

Start by typing `cmake --list-presets` to see which configure presets you can use. Choose one and then type `cmake --preset <name of the preset>`

## Build

Type `cmake --build --list-presets` to see which build presets you can use. Debug mode presets end with "debug". Release mode presets end with "release".

Then type `cmake --build --preset <name of the preset>`.

**Comment by RedNicStone** [comment]
This should absolutely be added to the `README.md`, I agree.
While the information is there, I should add how the `.appimage` is built.

- Make sure you have the prerequisite `appimagetool` installed.
- At configure time, configure with `-DENABLE_BUILD_APPIMAGE=True` to enable the appimage packager.
- Build as normal.
- Generate the AppImage using `cmake --install . --config [Debug/Release] --component desktop-shadertoy-appimage`.
- The generated binary is located in `[your-binary-dir]/appimage/desktop-shadertoy-[version]-[arch].appimage`.

**Comment by raki-huawei** [comment]
I'm trying to build the latest from master on windows using vs 2022 v17.13.6. 

>cmake --preset "cl-x86_64-windows-msvc"
>cmake --build --preset "cl-x86_64-windows-msvc-debug"

The build is failing with following errors

```
glslangd.lib(linkValidate.cpp.obj) : error LNK2019: unresolved external symbol __std_remove_8 referenced in function "class TIntermNode * * __cdecl std::_Remove_vectorized<class TIntermNode *,std::nullptr_t>(class TIntermNode * * const,class TIntermNode * * const,std::nullptr_t)" (??$_Remove_vectorized@PEAVTIntermNode@@$$T@std@@YAPEAPEAVTIntermNode@@QEAPEAV1@0$$T@Z)
glslangd.lib(ParseHelper.cpp.obj) : error LNK2019: unresolved external symbol __std_search_1 referenced in function "char const * __cdecl std::_Search_vectorized<char const ,char const >(char const * const,char const * const,char const * const,unsigned __int64)" (??$_Search_vectorized@$$CBD$$CBD@std@@YAPEBDQEBD00_K@Z)
glslangd.lib(propagateNoContraction.cpp.obj) : error LNK2001: unresolved external symbol __std_search_1
glslangd.lib(SpvBuilder.cpp.obj) : error LNK2019: unresolved external symbol __std_find_last_trivial_1 referenced in function "char const * __cdecl std::_Find_last_vectorized<char const ,char>(char const * const,char const * const,char)" (??$_Find_last_vectorized@$$CBDD@std@@YAPEBDQEBD0D@Z)
Debug\desktop-shadertoy.exe : fatal error LNK1120: 3 unresolved externals
```

Seems the culprit is glslang built with a different version of C++. I'm not sure. Any suggestions?

Thank you.
